nsxt/resource_nsxt_policy_ip_block_quota_test.go

```diff
@@ -0,0 +1,254 @@
+					resource.TestCheckResourceAttr(testResourceName, "quota.#", "1"),
+					resource.TestCheckResourceAttr(testResourceName, "quota.0.other_cidrs.#", "1"),
+					resource.TestCheckResourceAttrSet(testResourceName, "nsx_id"),
+					resource.TestCheckResourceAttrSet(testResourceName, "path"),
+					resource.TestCheckResourceAttrSet(testResourceName, "revision"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceNsxtPolicyIpBlockQuota_importBasic(t *testing.T) {
+	name := getAccTestResourceName()
+	testResourceName := "nsxt_policy_ip_block_quota.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtPolicyIpBlockQuotaCheckDestroy(state, name)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNsxtPolicyIpBlockQuotaMinimalistic(false),
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccNsxtPolicyIpBlockQuotaExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy IpBlockQuota resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("Policy IpBlockQuota resource ID not set in resources")
+		}
+
+		exists, err := resourceNsxtPolicyIpBlockQuotaExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("Policy IpBlockQuota %s does not exist", resourceID)
+		}
+
+		return nil
+	}
+}
+
+func testAccNsxtPolicyIpBlockQuotaCheckDestroy(state *terraform.State, displayName string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_policy_ip_block_quota" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		exists, err := resourceNsxtPolicyIpBlockQuotaExists(testAccGetSessionProjectContext(), resourceID, connector)
+		if err == nil {
+			return err
+		}
+
+		if exists {
+			return fmt.Errorf("Policy IpBlockQuota %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtPolicyIpBlockQuotaTemplate(createFlow bool, withContext bool) string {
+	var attrMap map[string]string
+	var ipblock string
+	if createFlow {
+		attrMap = accTestPolicyIpBlockQuotaCreateAttributes
+		ipblock = "test-ipv4"
+	} else {
+		attrMap = accTestPolicyIpBlockQuotaUpdateAttributes
+		ipblock = "test-ipv6"
+	}
+	context := ""
+	if withContext {
+		context = testAccNsxtPolicyMultitenancyContext()
+	}
+	return fmt.Sprintf(`
+resource "nsxt_policy_ip_block" "test-ipv4" {
```

Comment:
As commented above, it might be worth creating blocks with external visibility for these tests